contracts/leverage_trading/src/view.rs

```diff
@@ -320,78 +320,118 @@ impl Contract {
+                            side: OrderType::Buy,
+                            price: WBigDecimal::from(order.open_price),
+                            amount: U128(order.amount),
+                            filled: 0,
+                            total: LowU128::from(total),
+                        })
+                    }
+                    false => None,
+                }
+            })
+            .collect::<Vec<LimitOrderView>>();
+
+        pending_limit_orders.sort_by(|a, b| a.time_stamp.cmp(&b.time_stamp));
 
-        vec![limit_order_1, limit_order_2, limit_order_3]
+        let total_orders = U128(pending_limit_orders.len() as u128);
+
+        let sort_pending_orders = pending_limit_orders
+            .into_iter()
+            .skip((orders_per_page.0 * page.0 - orders_per_page.0) as usize)
+            .take(orders_per_page.0 as usize)
+            .collect();
+
+        LimitOrders {
+            data: sort_pending_orders,
+            page,
+            total_orders,
+        }
     }
 
-    #[allow(unused_variables)]
     pub fn view_pending_limit_orders_by_user_by_pair(
         &self,
         account_id: AccountId,
         sell_token: AccountId,
         buy_token: AccountId,
-    ) -> Vec<LimitOrderView> {
-        let order_type = "Limit".to_string();
-
-        let limit_order_1 = LimitOrderView {
-            time_stamp: 1675332394281,
-            pair: "USDT/NEAR".to_string(),
-            order_type: order_type.clone(),
-            side: OrderType::Buy,
-            price: WBalance::from(302 * 10_u128.pow(22)), // 3.05 / 1.01 = 3.01980198
-            amount: U128(500 * 10_u128.pow(24)),
-            filled: 0,
-            total: WBalance::from(505 * 10_u128.pow(24)), //500 * 1.01 = 505
-        };
+        orders_per_page: U128,
+        page: U128,
+    ) -> LimitOrders {
+        let orders = self.orders.get(&account_id).unwrap_or_default();
 
-        let limit_order_2 = LimitOrderView {
-            time_stamp: 1675336439706,
-            pair: "USDT/NEAR".to_string(),
-            order_type,
-            side: OrderType::Buy,
-            price: WBalance::from(302 * 10_u128.pow(22)), // 3.05 / 1.01 = 3.01980198
-            amount: U128(1000 * 10_u128.pow(24)),
-            filled: 0,
-            total: WBalance::from(1010 * 10_u128.pow(24)), //1000 * 1.01 = 1010
-        };
+        let mut pending_limit_orders = orders
+            .iter()
+            .filter_map(|(_, order)| {
+                match order.status == OrderStatus::Pending
+                    && order.leverage == BigDecimal::one()
+                    && order.sell_token == sell_token
+                    && order.buy_token == buy_token
+                {
+                    true => {
+                        let trade_pair = self.view_pair(&order.sell_token, &order.buy_token);
+
+                        let pair =
+                            format!("{}/{}", trade_pair.sell_ticker_id, trade_pair.buy_ticker_id);
```

Comment:
format!("{}/{}", trade_pair.sell_ticker_id, trade_pair.buy_ticker_id); => format!("{trade_pair.sell_ticker_id,}/{trade_pair.buy_ticker_id}");

Reply:
This is not possible in this form. A placeholder can only take a variable name. I leave it unchanged since the current format, in my opinion, is fine. But you can change it as follows:

1. **Positional arguments**
    `format!("{0}/{1}", trade_pair.sell_ticker_id, trade_pair.buy_ticker_id);`
2. **Named arguments**
    `format!("{sell_ticker_id}/{buy_ticker_id}", sell_ticker_id = trade_pair.sell_ticker_id, buy_ticker_id = 
     trade_pair.buy_ticker_id);` 
4. **Create two more variables**
    `let sell_ticker_id = trade_pair.sell_ticker_id;`
    `let buy_ticker_id = trade_pair.buy_ticker_id;`
    `format!("{sell_ticker_id}/{buy_ticker_id}");`